### PR TITLE
Remaining instruction tests for riscv.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: riscv/tests/instructions.rs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+use compiler::compile_asm_string_temp;
+use riscv::compiler::compile_riscv_asm;
+
+fn run_instruction_test(assembly: &str, name: &str) {
+    // TODO Should we create one powdr asm from all tests or keep them separate?
```
Comment (leonardoalt) [inline riscv/tests/instructions.rs:5]: Probably separated?

Comment (chriseth) [reply riscv/tests/instructions.rs:5]: The problem is that it takes ages to run the tests in that case.

Comment (leonardoalt) [reply riscv/tests/instructions.rs:5]: but does it make sense to have them all in a single asm file? how do you guarantee reachability of each test?

**Comment by chriseth** [comment]
I'll see to get some tests running and split them out into their own PR so we can merge it.

**Comment by leonardoalt** [comment]
Rebased

**Comment by leonardoalt** [comment]
Ok there are still multiple things missing

**Comment by chriseth** [comment]
lw and sw can be extracted, they are the only ones that work.

**Comment by leonardoalt** [comment]
rebased

**Comment by Schaeff** [comment]
Is there a path to this being merged?

**Comment by leonardoalt** [comment]
> Is there a path to this being merged?

Kinda. We pick chunks of this PR and merge it as smaller PRs. Eventually it will be small enough to be the last one.

**Comment by chriseth** [comment]
This essentially has the sources for the instruction tests.

**Pull request closed without merging**